src/django_components/util/html.py

```diff
@@ -1,100 +1,108 @@
-from typing import List, Union
+from abc import ABC, abstractmethod
+from typing import Any, List, Optional, Sequence
 
-from selectolax.lexbor import LexborHTMLParser, LexborNode
+from bs4 import BeautifulSoup, CData, Comment, Doctype, Tag
 
 
-def parse_node(html: str) -> LexborNode:
+class HTMLNode(ABC):
```

Comment:
Since I changed from Selectolax to lxml to BS4, I defined the interface for the HTML parser.

<details>
<summary>
E.g. click here to see implementation for lxml
</summary>

```py
class LXMLNode(HTMLNode):
    def __init__(self, node: lxml.etree.ElementBase):
        self.node = node

    @classmethod
    def from_fragment(self, html: str) -> List["LXMLNode"]:
        elems = lxml.html.fragments_fromstring(html)
        return [LXMLNode(elem) for elem in elems]

    def to_html(self) -> str:
        return lxml.html.tostring(self.node, encoding="unicode")

    def find_tag(self, tag: str) -> Optional["LXMLNode"]:
        if self.node.tag == tag:
            return self
        else:
            matches = self.node.xpath(tag)
            if matches:
                return LXMLNode(matches[0])

        return None

    def append_children(self, children: Sequence["LXMLNode"]) -> None:
        for child in children:
            self.node.append(child.node)

    def get_attr(self, attr: str, default: Any = None) -> Any:
        return self.node.get(attr, default)

    def set_attr(self, attr: str, value: Any) -> None:
        if value is True:
            self.node.set(attr, None)
        elif value is False:
            self.node.attrib.pop(attr, None)
        else:
            self.node.set(attr, value)

    def is_element(self) -> bool:
        return isinstance(self.node, lxml.html.HtmlElement)
```

</details>

I also added a test for this interface, in case we want to change the implementation in the future.

That's because one thing that I keep in mind is that if in the future we get to measure performance for bottlenecks, and it turns out that HTML parsing is a bottleneck, then I have written a lightweight HTML parser for django-vue. So in case of a bottleneck, that parser could be migrated to Cython, and then it should have similar performance as lxml or Selectolax.